core/bench_contract_test.go

```diff
@@ -0,0 +1,87 @@
+// (c) 2020-2021, Ava Labs, Inc.
+//
+// This file is a derived work, based on the go-ethereum library whose original
+// notices appear below.
+//
+// It is distributed under a license compatible with the licensing terms of the
+// original code from which it is derived.
+//
+// Much love to the original authors for their work.
+// **********
+// Copyright 2014 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	_ "embed"
+	"math/big"
+	"testing"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var (
+	//go:embed TrieStressTest.bin
+	stressBinStr string
+	//go:embed TrieStressTest.abi
+	stressABIStr string
+)
+
+func BenchmarkTrie(t *testing.B) {
+	benchInsertChain(t, true, stressTestTrieDb(100, 100, 5000))
+}
+
+func stressTestTrieDb(numContracts int, callsPerBlock int, elements int64) func(int, *BlockGen) {
+	contractAddr := make([]common.Address, numContracts)
+	contractTxs := make([]*types.Transaction, numContracts)
+
+	gasPrice := big.NewInt(225000000000)
+	gasTx := uint64(22000)
```

Comment:
If you specify this amount of gas, then regardless of how many elements are requested to be modified once it hits this limit it will hit `ErrOutOfGas` and revert.

Could we change the logic here to make sure that it sanity checks it has enough gas to execute the requested number of elements?

Reply:
We have a better error reporting system now, it will panic/error if ErrOutGas is returned. Is that ok @aaronbuchwald ?